circuits/src/zk_gadgets/wallet_operations.rs

```diff
@@ -0,0 +1,211 @@
+//! Groups logic for computing wallet commitments and nullifiers inside of a circuit
+
+use mpc_bulletproof::{
+    r1cs::{LinearCombination, RandomizableConstraintSystem, Variable},
+    r1cs_mpc::R1CSError,
+};
+
+use crate::{
+    mpc_gadgets::poseidon::PoseidonSpongeParameters,
+    types::{balance::BalanceVar, fee::FeeVar, order::OrderVar, wallet::WalletSecretShareVar},
+};
+
+use super::{comparators::EqVecGadget, poseidon::PoseidonHashGadget};
+
+// ------------------------
+// | Public State Gadgets |
```

Comment:
The public state gadgets are unchanged from before, I added the wallet state primitive comparators, but the diff seems to suggest the whole file changed.